### ISSUE TITLE
Help wanted: "Could not create new database" and "Couldn't access conflicts view" (Docker)

### ISSUE THREAD

**Issue by Jonny1797** [davideshay/groceries#110]
Hi, 

I was just setting up a new instance with docker compose according to the docs to test this great project. Unfortunately, I hit a wall in what seems like a combination of issue #108 and #92. 

At startup, I always see 
```
backend-groceries-server   | [17:08:50] INFO: Starting up auth server for couchdb...
backend-groceries-server   | [17:08:50] INFO: App Version:  1.1.0
backend-groceries-server   | [17:08:50] INFO: Database Schema Version: 6
backend-groceries-server   | [17:08:50] INFO: Database URL:  https://couchdb.mytld.com
backend-groceries-server   | [17:08:50] INFO: Internal Database URL:  http://couchdb:5984
backend-groceries-server   | [17:08:50] INFO: Using database: groceries
backend-groceries-server   | [17:08:50] INFO: Refresh token expires in  30d(2592000 seconds)
backend-groceries-server   | [17:08:50] INFO: STATUS: Access token expires in  12h(43200 seconds)
backend-groceries-server   | [17:08:50] INFO: STATUS: User Account Creation is:  ENABLED
backend-groceries-server   | [17:08:50] ERROR: could not retrieve database info.
backend-groceries-server   | [17:08:50] INFO: Database does not exist... Creating....
backend-groceries-server   | [17:08:50] ERROR: Could not create new database
backend-groceries-server   | [17:08:50] INFO: Database existence checked/created.
backend-groceries-server   | ---ERROR IN STARTUP--- EXITING BACKEND
```

and later on the huge JSON output which begins with `backend-groceries-server   | [17:06:06] ERROR: Couldn't access conflicts view: Error: missing_named_view`.

I used the standard docker-compse.yaml in which I altered the `COUCHDB_HMAC_KEY` and according to that I changed the jwt.ini.
(BTW I deleted an " at the end of `COUCHDB_ADMIN_USER` which I supposed shouldn't be there. Source is the link from the documentation. On GitHub, this file seems fine.) Database was available at http://\<ip\>:5984. I was able to log in but, maybe expected, at http://\<ip\>:5984/groceries/_design/_conflicts_only_view_id/_view/conflicts_view the response was `{"error":"not_found","reason":"missing_named_view"}`.

Unfortunately, these errors give me no clue for what I can do to make it work. All help is appreciated.

**Comment by davideshay** [comment]
Can you post or message me your full docker compose yaml file? Also, not sure if it will help, but can you set the logging on the backend to debug mode?  

I'll see if I can add any more logging on the "could not create new database" because I think that's where the problem probably is. Did you create the database in couchdb in advance or are you letting the app create it?

**Comment by davideshay** [comment]
Also would be good to see the other logs from docker compose up, especially to see if couchdb had any issues getting started. And the default docker-compose file expects both a dbdata directory to be created as well as a dbetclocal directory to be created, with both an "admin.ini" file with username/password and a "jwt.ini" file with the HMAC key. Do those files exist?

**Comment by Jonny1797** [comment]
First, can you please provide the information how to enable debug logging, as I haven't found a way to do that.
Second, the database is created by the app, which seems to work as the logs indicate. :thinking: 
Furthermore, all the files and folders exist like in the [docker-full.tar.gz](https://raw.githubusercontent.com/davideshay/groceries/master/mkdocs/docs/assets/docker-full.tar.gz).
Here is the log output of docker compose up: [logs.log](https://github.com/davideshay/groceries/files/13611982/logs.log)
Here is my docker-compose.yaml: [docker-compose.yaml.txt](https://github.com/davideshay/groceries/files/13612139/docker-compose.yaml.txt) (I got my own caddy on another machine, so that is missing on purpose.)
And here is the altered jwt.ini: [jwt.ini.txt](https://github.com/davideshay/groceries/files/13612210/jwt.ini.txt). 
All other files are left in their default state.

Let me know if you need anything else.

(Side note: correct me if I'm wrong, but empty directories don't have to exist in advance, because docker compose will create them, right?)

**Comment by davideshay** [comment]
For reference, but the frontend and backend components take an environment variable LOG_LEVEL that can be set to, for instance, "D" or "1" to enable debug logging. I am updating the docs to make this clear.

Also, not sure if you are just obfuscating your files, which is fine, but you will need to change "couchdb.mytld.com" to your actual externally accessible (outside the docker machine/cluster) FQDN that caddy is serving up. Same with the references to "groceries.mytld.com".

**Comment by Jonny1797** [comment]
Of course, the problem again sits in front of the PC. I took the docker-compose.yaml for granted so that I didn't check the versioning :astonished:.

Yes, I got the `admin.ini` at its right place but with default contents, so I didn't past it in here.

All right, thank you about updating the docs, I wasn't aware of this environment variable.

I'm aware that I had to change the "*.mytld.com" variables, but the docs indicated that one should do this at a later point, where I didn't get yet.

So *Specifically Clementines* is up and running now, thank you! I think the really confusing part is that the link to the [docker-full.tar.gz](https://raw.githubusercontent.com/davideshay/groceries/master/mkdocs/docs/assets/docker-full.tar.gz) contains outdated / wrong information. This includes the versions of the containers (both back and frontend) as well as at least one variable (`admin"` instead of `admin`). New users may not be aware of this.

Thank you again for taking the time to resolve this!
